sdks/nodejs/src/agonesSDK.js

```diff
@@ -0,0 +1,101 @@
+const grpc = require('grpc');
+
+const messages = require('../lib/sdk_pb');
+const services = require('../lib/sdk_grpc_pb');
+
+class AgonesSDK {
+	constructor() {
+		this.client = new services.SDKClient('localhost:50051', grpc.credentials.createInsecure());
+	}
+
+	async ready() {
+		let request = new messages.Empty();
+		return new Promise((resolve, reject) => {
+			this.client.ready(request, (error, response) => {
+				if (error) {
+					reject(error);
+				} else {
+					resolve(response.toObject());
+				}
+			});
+		});
+	}
+
+	async shutdown() {
+		let request = new messages.Empty();
+		return new Promise((resolve, reject) => {
+			this.client.shutdown(request, (error, response) => {
+				if (error) {
+					reject(error);
+				} else {
+					resolve(response.toObject());
+				}
+			});
+		});
+	}
+
+	health() {
+		if (!this.healthStream) {
+			this.healthStream = this.client.health((error) => {
+				if (error) {
+					throw error;
+				}
+			});
+		}
+		let request = new messages.Empty();
+		this.healthStream.write(request);
+	}
+
+	async getGameServer() {
+		let request = new messages.Empty();
+		return new Promise((resolve, reject) => {
+			this.client.getGameServer(request, (error, response) => {
+				if (error) {
+					reject(error);
+				} else {
+					resolve(response.toObject());
+				}
+			});
+		});
+	}
+
+	watchGameServer(callback) {
+		let request = new messages.Empty();
+		let emitter = this.client.watchGameServer(request);
+		emitter.on('data', (data) => {
```

Comment:
I do not handle the 'end' event as not sure what to do with it and did not see it fired during testing